src/dump.c

```diff
@@ -32,7 +32,7 @@
 #include <string.h>
 #include <math.h>
 
-#ifdef __EMBEDDED__
+#ifdef STM32F40_41xxx
 #include "gdb_stdio.h"
```

Comment:
It would be nice to avoid an arch-dep conditionals.

Is the issue here that we need to differentiate between sm1000/stm32 and other embedded/small platforms?

Reply:
Correct. gdb_stdio.h from what I can tell is only in the build environment used for SM1000 builds.

Reply:
OK this isn't needed now that https://github.com/drowe67/codec2/pull/247 is merged.